src/vite/island-components.ts

```diff
@@ -52,6 +52,18 @@ function isComponentName(name: string) {
   return /^[A-Z][A-Z0-9]*[a-z][A-Za-z0-9]*$/.test(name)
 }
 
+/**
+ * Matches when id is the filename of Island component
+ *
+ * @param id - The id to match
+ * @returns The result object if id is matched or null
+ */
+export function matchIslandComponentId(id: string) {
+  return id.match(
+    /(\/islands\/.+?\.tsx$)|(\/routes\/.*\_[a-zA-Z0-9[-]+\.island\.tsx$)|(\/routes.*\/\$[a-zA-Z0-9[-]+\.tsx$)/
```

Comment:
Is it intended that there is no `/` under `routes` ? If not, I think it should be added.

```suggestion
    /(\/islands\/.+?\.tsx$)|(\/routes\/.*\_[a-zA-Z0-9[-]+\.island\.tsx$)|(\/routes\/.*\/\$[a-zA-Z0-9[-]+\.tsx$)/
```

We might also want to take this opportunity to review our existing regular expressions as well. I think I can summarize as follows

```ts
  return id.match(
    /\/islands\/.+?\.tsx$|\/routes\/(?:.*\/)?(?:\_[a-zA-Z0-9-]+\.island\.tsx$|\$[a-zA-Z0-9-]+\.tsx$)/
  )
```

![CleanShot 2024-05-09 at 05 48 45@2x](https://github.com/honojs/honox/assets/30598/cd38b69f-2ed5-452c-9d2a-0bc474d24cb0)

Reply:
> Is it intended that there is no `/` under `routes` ? If not, I think it should be added.

I'd like it to match `/routes/$counter.tsx`.

> Is `[` necessary?

Exactly, no! It's not necessary. It's (maybe my) fault.

```ts
  return id.match(
    /\/islands\/.+?\.tsx$|\/routes\/(?:.*\/)?(?:\_[a-zA-Z0-9-]+\.island\.tsx$|\$[a-zA-Z0-9-]+\.tsx$)/
  )
```

Great! This will match `/routes/$counter.tsx`. I'll change the regex to yours and add some tests. I'll also add you as a co-author for this PR.

Reply:
👍